CONTRIBUTING.md

```diff
@@ -0,0 +1,505 @@
+debian/control file in this repository. You can build the
+packages the way you would normally build a Debian package:
+
+
+```shell
+dpkg-buildpackage -us -uc
+```
+
+**Note** It will build the packages with dependencies for the Ubuntu release on
+which you are building, so it's best to build in a container or kvm for the
+release you are targeting.
+
+OR, if you want to build for a target release other than the release
+you're on:
+
+### using sbuild
+[configure sbuild](https://wiki.ubuntu.com/SimpleSbuild) and
+use that for the build:
+
+Setup some chroots for sbuild with this script
+```shell
+bash ./tools/setup_sbuild.sh
+```
+
+```shell
+debuild -S
+sbuild --dist=<target> ../ubuntu-advantage-tools_*.dsc
+# emulating different architectures in sbuild-launchpad-chroot
+sbuild-launchpad-chroot create --architecture="riscv64" "--name=focal-riscv64" "--series=focal
+```
+
+> Note: Every so often, it is recommended to update your chroots.
+> ```bash
+> # to update a single chroot
+> sudo sbuild-launchpad-chroot update -n ua-xenial-amd64
+> # this script can be used to update all chroots
+> sudo PATTERN=\* sh /usr/share/doc/sbuild/examples/sbuild-debian-developer-setup-update-all
+> ```
+
+### Setting up an lxc development container
+```shell
+lxc launch ubuntu-daily:trusty dev-t -c user.user-data="$(cat tools/ua-dev-cloud-config.yaml)"
+lxc exec dev-t bash
+```
+
+### Setting up a kvm development environment with multipass
+**Note:** There is a sample procedure documented in tools/multipass.md as well.
+```shell
+multipass launch daily:focal -n dev-f --cloud-init tools/ua-dev-cloud-config.yaml
+multipass connect dev-f
+```
+
+## Code Formatting
+
+The `ubuntu-advantage-client` code base is formatted using
+[black](https://github.com/psf/black), and imports are sorted with
+[isort](https://github.com/PyCQA/isort).  When making changes, you
+should ensure that your code is blackened and isorted, or it will
+be rejected by CI.
+Formatting the whole codebase is as simple as running:
+
+```shell
+black uaclient/
+isort uaclient/
+```
+
+To make it easier to avoid committing incorrectly formatted code, this
+repo includes configuration for [pre-commit](https://pre-commit.com/)
+which will stop you from committing any code that isn't blackened.  To
+install the project's pre-commit hook, install `pre-commit` and run:
+
+```shell
+pre-commit install
+```
+
+(To install `black` and `pre-commit` at the appropriate versions for
+the project, you should install them via `dev-requirements.txt`.)
+
+## Daily Builds
+
+On Launchpad, there is a [daily build recipe](https://code.launchpad.net/~canonical-server/+recipe/ua-client-daily),
+which will build the client and place it in the [ua-client-daily PPA](https://code.launchpad.net/~ua-client/+archive/ubuntu/daily).
+
+## Releasing ubuntu-advantage-tools
+See [How to release a new version of UA](./contributing-docs/howtoguides/how_to_release_a_new_version_of_ua.md)
```

Comment:
I think the contributing file is nice! I'd just add information about how to send a pull request; let's say I want to contribute, then I read this guide, perform the change, run the tests and all - I'd like to know the next steps: clone the repo, send a PR? Sign the Ubuntu CLA / Contributor guide? Who should approve this PR? These things would help I guess

Reply:
Totally agree! I only intended this PR to be the reorganization to separate the existing developer docs from the user docs - then in future PRs we could iterate on the content.

So I agree that we should have that content but it feels like a separate PR after we agree on the basic organization in this PR